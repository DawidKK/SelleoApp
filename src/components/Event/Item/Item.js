import React from 'react';

import EventItemStyle from './ItemStyles';

const EventItem = ({ event }) => {
  const { title, description } = event;

  return (
    <li>
      <EventItemStyle>
        <p className="event-item__title">{title}</p>
        <p className="event-item__description">{description}</p>
      </EventItemStyle>
    </li>
  );
};

export default EventItem;
