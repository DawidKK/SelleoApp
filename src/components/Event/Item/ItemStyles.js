import styled from 'styled-components';

const EventItemStyle = styled.div`
  border: 1px solid ${props => props.theme.accentColor};
  padding: 1rem;
  .event-item__title {
    font-size: 1.7rem;
    margin: 0;
  }

  .event-item__description {
    background: ${props => props.theme.bgLight};
    padding: 10px;
  }
`;

export default EventItemStyle;
