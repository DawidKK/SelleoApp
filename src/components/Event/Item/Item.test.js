import React from 'react';
import Item from './Item';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

const fakeItem = {
  title: 'test',
  description: 'test description'
};

describe('<Item />', () => {
  it('renders Item properly', () => {
    const wrapper = shallow(<Item event={fakeItem} />);
    const title = wrapper.find('.event-item__title');
    const description = wrapper.find('.event-item__description');

    expect(title.text()).toBe('test');
    expect(description.text()).toBe('test description');
  });

  it('matches snapshot', () => {
    const wrapper = shallow(<Item event={fakeItem} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
