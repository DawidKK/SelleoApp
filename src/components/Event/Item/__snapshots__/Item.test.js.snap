// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item /> matches snapshot 1`] = `
<li>
  <styled.div>
    <p
      className="event-item__title"
    >
      test
    </p>
    <p
      className="event-item__description"
    >
      test description
    </p>
  </styled.div>
</li>
`;
